discovery-infra/test_infra/tools/connection.py

```diff
@@ -0,0 +1,17 @@
+import socket
+import logging
+
+
+def is_open(ip, port, timeout=5):
+    s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    s.settimeout(timeout)
+    try:
+        s.connect((ip, int(port)))
+        s.shutdown(socket.SHUT_RDWR)
+        logging.info("Ip %s with port %s are reachable", ip, port)
+        return True
+    except:
```

Comment:
Set specific exceptions

Reply:
no need, it is fine to fall on any error, don't want to handle here any logic

Reply:
If you provided something strange, like a wrong name `NameError` would be raised. Other examples like `RecursionError`, `RunTimeError`, `IndexError`.. - All of those you want to know about.
You should explicitly choose the accepted exceptions. I would suggest even to take the whole socket errors family `socket.error` (`socket.timeout` is a subclass of `socket.error`)